azure/interfaces.go

```diff
@@ -106,6 +112,8 @@ type ResourceSpecGetter interface {
 	OwnerResourceName() string
 	// ResourceGroupName returns the name of the resource group the resource is in.
 	ResourceGroupName() string
+	// CustomHeaders returns the headers that should be added to Azure API calls.
+	CustomHeaders() map[string]string
```

Comment:
@alexeldeib had also suggested splitting it out. I've added a commit to add a separate interface that has `CustomHeaders()`. The downside is that it required casting in client.go before using it but it's a little cleaner since it doesn't require touching any of the other services. Let me know what you think.

Reply:
Yes. In separate interface -- it looks much cleaner. This should be fine to go with the PR.